builder/builder.go

```diff
@@ -1,4 +1,4 @@
-package tasks
+package builder
```

Comment:
The common code is now a bit all over the place.

- There's some code still in `tasks` package
- There's some under `tasks/shared`
- And the builder is here under root.

I would make `tasks` folder only container for different task packages, and move all common code either in root, or some other subfolder of root.

Reply:
yes, i'd noticed this too. for now, what i've done is move the `builder` to the `tasks/shared` folder, and then created a `tasks/support` folder as well and moved the specific task support packages there. the other root folders feel about right and the moved packages feels like they belong under `tasks`. not quite sure whether we need `shared` and `support` but i'll leave for now and a future refactoring can consolidate if better.

btw, i think getting more of the common code into their own packages will also help with the circular dependencies (see below) but need to draw a line somewhere for this set of changes.